File: data/src/main/java/dev/olog/data/api/lastfm/track/TrackSearch.java

package dev.olog.data.api.lastfm.track;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class TrackSearch {

    @Expose
    public Results results;

    public static class Results {

        @Expose
        public Trackmatches trackmatches;

        public static class Trackmatches {

            @Expose
            public List<Track> track = new ArrayList<Track>();

            public static class Track {

                @Expose
                public String name;

                @Expose
                public String artist;

                @Expose
                public String url;

                @Expose
                public List<Image> image = new ArrayList<Image>();

                @Expose
                public String mbid;

                public static class Image {

                    @SerializedName("#text")
                    @Expose
                    public String text;
                    @Expose
                    public String size;

                }

            }

        }

    }

}

File: data/src/main/java/dev/olog/data/api/lastfm/track/TrackSearch.kt
package dev.olog.data.api.lastfm.track

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class TrackSearch(
    val results: Results? = null,
) {

    @Serializable
    class Results(
        val trackmatches: Trackmatches? = null,
    )

    @Serializable
    class Trackmatches(
        val track: List<Track>? = null
    )

    @Serializable
    class Track(
        val name: String? = null,
        val artist: String? = null,
        val url: String? = null,
        val image: List<Image>? = null,
        val mbid: String? = null
    )

    @Serializable
    class Image(
        @SerialName("#text")
        val text: String? = null,
        val size: String? = null,
    )

}